ugc/binary/serdes.cpp

```diff
@@ -0,0 +1,109 @@
+#include "ugc/binary/serdes.hpp"
+
+#include <set>
+
+using namespace std;
+
+namespace ugc
+{
+namespace binary
+{
+namespace
+{
+class BaseCollector
+{
+public:
+  virtual void VisitRating(float const f, char const * /* name */ = nullptr) {}
```

Comment:
There's no real need in virtual dtor, as the instances won't be passed or destructed polymorphically.